### PR TITLE
feat: Support 'go to test' command

### PR DESCRIPTION
**Pull request by jdneo**
part of #660

Depending on the search result, this feature has following response:

1. No tests found

https://user-images.githubusercontent.com/6193897/141043353-c27ddc35-92fa-4b44-825e-6a9e3b149c6f.mp4

2. One match test found

https://user-images.githubusercontent.com/6193897/141043306-e226664b-972a-4e66-979c-8743f20c0613.mp4

3. Multiple tests found

https://user-images.githubusercontent.com/6193897/141043323-65338679-9360-465b-b722-df7a83c1eaa1.mp4

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        if (arguments == null || arguments.size() < 2) {
+            throw new IllegalArgumentException("Wrong arguments passed to findTestOrTarget().");
+        }
+        final String typeUri = (String) arguments.get(0);
+        final ICompilationUnit unit = JDTUtils.resolveCompilationUnit(typeUri);
+        if (unit == null) {
+            JUnitPlugin.logError("Failed to resolve compilation unit from " + typeUri);
+            return null;
+        }
+        final boolean goToTest = (boolean) arguments.get(1);
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java:57]: what's "false" reserved for?

File: src/commands/navigationCommands.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        return;
+    }
+    const uri: Uri = window.activeTextEditor.document.uri;
+    if (await testSourceProvider.isOnTestSourcePath(uri)) {
+        return;
+    }
+    const results: ITestFindResult[] = await searchTests(uri.toString());
+    if (!results.length) {
+        window.showQuickPick([
+            'Generate tests...',
```
Comment (Eskibear) [inline src/commands/navigationCommands.ts:21]: extract options to variables?

**Review by Eskibear — DISMISSED** [review]

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java (hunk 1 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java:57]: It's reserved for `test code -> business code`

File: src/commands/navigationCommands.ts (hunk 2 above)
Comment (jdneo) [reply src/commands/navigationCommands.ts]: Done

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            new TestNavigationNameRequestor(items, javaProject, typeName),
+            IJavaSearchConstants.WAIT_UNTIL_READY_TO_SEARCH,
+            monitor
+        );
+
+        return new TestNavigationResult(items, location);
+    }
+
+    private static IJavaSearchScope getSearchScopeForTarget() {
+        // TODO: unimplemented
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java:95]: won't be implemented in this PR? any issue to track?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            final String fullyQualifiedName = String.valueOf(packageName) + "." + simpleName;
+            int relevance;
+            if (Objects.equals(simpleName, this.typeName + "Test") ||
+                    Objects.equals(simpleName, this.typeName + "Tests")) {
+                // mark this as most relevance
+                relevance = Integer.MIN_VALUE;
+            } else {
+                relevance = simpleName.indexOf(this.typeName);
+                if (relevance < 0) {
+                    relevance = 1000;
```
Comment (Eskibear) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java:154]: magic number 1000? how about INT_MAX?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java (hunk 3 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java:95]: There will be a separate PR. Tracked by https://github.com/microsoft/vscode-java-test/issues/660

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java (hunk 4 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java]: Use INT_MAX now.

**Review by Eskibear — APPROVED** [review]
LGTM. remember to track todo item.

**Pull request merged**